### PR TITLE
[Spec] Ensures for voting

### PR DESCRIPTION
**Pull request by UIZorrot**
### Description

We have finished the remained post-condition in voting.move.
As far as we know, high-level properties is all complete

### Test Plan

aptos move prove -f voting

### CODE REVIEW

**Review by rahxephon89 — CHANGES_REQUESTED** [review]
@UIZorrot I think `resolve_proposal_v2` will cause inconsistency issue. Could you use the flag  `--unconditional-abort-as-inconsistency` and `--check-inconsistency` to check whether there is an issue?

**Comment by UIZorrot** [comment]
The inconsistency is coming from event.move, the latest updated fix the problem

**Review by rahxephon89 — CHANGES_REQUESTED** [review]

File: aptos-move/framework/aptos-framework/sources/voting.spec.move [hunk 1]
```diff
@@ -57,15 +55,11 @@ spec aptos_framework::voting {
         is_multi_step_proposal: bool,
     ): u64 {
         use aptos_framework::chain_status;
-        pragma verify_duration_estimate = 120; // TODO: set because of timeout (property proved)
 
         requires chain_status::is_operating();
         include CreateProposalAbortsIf<ProposalType>;
+        // property 1: Verify the proposal_id of the newly created proposal.
         ensures result == old(global<VotingForum<ProposalType>>(voting_forum_address)).next_proposal_id;
-        ensures global<VotingForum<ProposalType>>(voting_forum_address).next_proposal_id
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/voting.spec.move:65]: Why are these removed?

Comment (rahxephon89) [reply aptos-move/framework/aptos-framework/sources/voting.spec.move:65]: They are moved to the schema below

**Review by rahxephon89 — CHANGES_REQUESTED** [review]

File: aptos-move/framework/aptos-framework/sources/voting.spec.move [hunk 2]
```diff
@@ -176,25 +211,70 @@ spec aptos_framework::voting {
...
+
+        let multi_step_key = std::string::spec_utf8(IS_MULTI_STEP_PROPOSAL_KEY);
+        aborts_if simple_map::spec_contains_key(proposal.metadata, multi_step_key) &&
+            !from_bcs::deserializable<bool>(simple_map::spec_get(proposal.metadata, multi_step_key));
+        let is_multi_step = simple_map::spec_contains_key(proposal.metadata, multi_step_key) &&
+            from_bcs::deserialize(simple_map::spec_get(proposal.metadata, multi_step_key));
+        aborts_if !is_multi_step && len(next_execution_hash) != 0;
+
+        aborts_if len(next_execution_hash) == 0 && !exists<timestamp::CurrentTimeMicroseconds>(@aptos_framework);
+        aborts_if len(next_execution_hash) == 0 && is_multi_step && simple_map::spec_contains_key(proposal.metadata, multi_step_in_execution_key);
```
[the remark this answers is not in the document]
Comment (UIZorrot) [reply aptos-move/framework/aptos-framework/sources/voting.spec.move:234]: oh yes, i think you are right

**Review by rahxephon89 — APPROVED** [review]

**Review by junkil-park — APPROVED** [review]

**Pull request merged**